spring-cloud-sleuth-core/src/test/java/org/springframework/cloud/sleuth/instrument/hystrix/JavanicaITest.java

```diff
@@ -0,0 +1,125 @@
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.cloud.netflix.hystrix.EnableHystrix;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.TraceManager;
+import org.springframework.cloud.sleuth.instrument.DefaultTestAutoConfiguration;
+import org.springframework.cloud.sleuth.trace.TraceContextHolder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.EnableAspectJAutoProxy;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+import com.jayway.awaitility.Awaitility;
+import com.netflix.hystrix.contrib.javanica.annotation.HystrixCommand;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = {
+		JavanicaITest.JavanicaITestConfiguration.class })
+public class JavanicaITest {
+
+	@Autowired JavanicaClass javanicaClass;
+	@Autowired JavanicaDelegation javanicaDelegation;
+	@Autowired TraceManager traceManager;
+
+	@Test
+	public void should_set_span_on_an_hystrix_command_annotated_method() {
+		final Span span = givenASpanInCurrentThread();
+
+		whenHystrixCommandGetsExecutedViaJavanica();
+
+		thenSpanPutInTheAsyncThreadIsSameAs(span);
+	}
+
+	private Span givenASpanInCurrentThread() {
+		Span span = this.traceManager.startSpan("existing").getSpan();
+		this.traceManager.continueSpan(span);
+		return span;
+	}
+
+	private void whenHystrixCommandGetsExecutedViaJavanica() {
+		this.javanicaDelegation.doSthThatDelegatesToJavanica();
+	}
+
+	private void thenSpanPutInTheAsyncThreadIsSameAs(final Span span) {
+		Awaitility.await().until(new Runnable() {
+			@Override
+			public void run() {
+				then(span.getTraceId()).isNotNull()
+						.isEqualTo(javanicaClass.getTraceId());
+				then(span.getName())
+						.isNotEqualTo(javanicaClass.getSpanName());
+			}
+		});
+	}
+
+	@After
+	public void cleanTrace() {
+		TraceContextHolder.removeCurrentTrace();
+	}
+
+	@DefaultTestAutoConfiguration
+	@EnableHystrix
+	@EnableAspectJAutoProxy(proxyTargetClass = true)
+	@Configuration
+	public static class JavanicaITestConfiguration {
+
+		@Bean
+		JavanicaClass javanicaClass() {
+			return new JavanicaClass();
+		}
+
+		@Bean
+		JavanicaDelegation javanicaDelegation() {
+			return new JavanicaDelegation(javanicaClass());
+		}
+	}
+
+	public static class JavanicaDelegation {
+
+		private final JavanicaClass javanicaClass;
+
+		public JavanicaDelegation(JavanicaClass javanicaClass) {
+			this.javanicaClass = javanicaClass;
+		}
+
+		public void doSthThatDelegatesToJavanica() {
+			this.javanicaClass.doSth();
+		}
+	}
+
+	public static class JavanicaClass {
+
+		AtomicReference<Span> span;
+
+		@HystrixCommand
+		public void doSth() {
```

Comment:
Good point - I'll refactor the names. I'm not a fan of adding javadocs where it's not necessary.

Instead of writing

```
 /** Catches the span associated with a hystrix command invocation. */
 static class JavanicaSpanCatcher {
```

I can just name the class

```
 static class HystrixCommandInvocationSpanCatcher {
```